### PR TITLE
[TESTING, CYPRESS] 3563 tests for the teacher features in the customize class page

### LINKED ISSUE #3563: [TESTING] Create front-end tests for the teacher features in the customize class page
**Description**
Create front-end tests in Cypress for all the features in the `/for-teachers/customize-class/ [class id]` page.

*Adventures:*
- the individual adventure/level buttons
- the checkboxes inside each adventures/level

*Opening dates:*
- Changing dates for every level via keyboard or button on the right

*select own adventures:*
- checkbox select adventure

*Other settings:*
- checkbox mandatory developer’s mode
- checkbox hide cheatsheet
- checkbox hide keyword switcher
- checkbox hide quiz
- checkbox hide puzzle

### PR DESCRIPTION
**Pull request by sharandaa**
**Description**

Adds cypress front end tests for all the features on the /for-teachers/customize-class/[class id] page. So, this includes the individual adventure buttons, opening dates, quiz score and all the checkboxes on the page. It is implemented in a dynamic fashion.

**Fixes _issue or discussion number_**

Fixes #3563 

**How to test**

Run 
`export ADMIN_USER=admin_user; python app.py` in your virtual environment. This ensures that there is an active admin_user and Hedy is running.

Then you can go to the /tests folder and run: 

`npx cypress run --spec cypress/e2e/teacher_page/customize_class_page`

### CODE REVIEW

File: tests/cypress/e2e/teacher_page/customize_class_page/ind_adv_button.cy.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+    cy.get('#class_view_button > .no-underline').click() // Press on view class button
+
+    cy.get('#customize-class-button').click(); // Press customize class button
+
+    // check every individual level button:
+    cy.get('#level_button_1').click()
+    cy.get('.adventure_level_1').should('be.not.checked')
+    cy.get('#level_button_1').click()
+    cy.get('.adventure_level_1').should('be.checked')
```
Comment (TiBiBa) [inline tests/cypress/e2e/teacher_page/customize_class_page/ind_adv_button.cy.js:18]: Is there a way to make this a bit more dynamic? This is a lot of duplicate code for all levels.

**Review by TiBiBa — CHANGES_REQUESTED** [review]
Left some comment, let me know if you need any help or have any questions!

File: tests/cypress/e2e/teacher_page/customize_class_page/test_checkboxes.cy.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    cy.get('#class_view_button > .no-underline').click() // Press on view class button
+
+    cy.get('#customize-class-button').click(); // Press customize class button
+    
+    // following code checks every single checkbox on the current page:
+    cy.get('[type="checkbox"]').check({force:true})
+    cy.get('[type="checkbox"]').should('be.checked')
+    cy.get('[type="checkbox"]').uncheck()
+    cy.get('[type="checkbox"]').should('be.not.checked')
```
[the remark this answers is not in the document]
Comment (sharandaa) [reply tests/cypress/e2e/teacher_page/customize_class_page/test_checkboxes.cy.js:18]: With the 'should' lines, it checks if the checkboxes are indeed all checked or unchecked.

File: tests/cypress/e2e/teacher_page/customize_class_page/opening_dates.cy.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    cy.get("#opening_date_level_9").type("2023-01-01").should("not.be.empty")
+    cy.get("#opening_date_level_10").type("2023-01-01").should("not.be.empty")
+    cy.get("#opening_date_level_11").type("2023-01-01").should("not.be.empty")
+    cy.get("#opening_date_level_12").type("2023-01-01").should("not.be.empty")
+    cy.get("#opening_date_level_13").type("2023-01-01").should("not.be.empty")
+    cy.get("#opening_date_level_14").type("2023-01-01").should("not.be.empty")
+    cy.get("#opening_date_level_15").type("2023-01-01").should("not.be.empty")
+    cy.get("#opening_date_level_16").type("2023-01-01").should("not.be.empty")
+    cy.get("#opening_date_level_17").type("2023-01-01").should("not.be.empty")
+    cy.get("#opening_date_level_18").type("2023-01-01").should("not.be.empty")
```
[the remark this answers is not in the document]
Comment (sharandaa) [reply tests/cypress/e2e/teacher_page/customize_class_page/opening_dates.cy.js:37]: I tried making it more dynamic by using this code: 
`cy.get('input[type=text]').type("2023-01-01").should("not.be.empty")`
However, I get this error:
`cy.type() can only be called on a single element. Your subject contained 19 elements.`
So, I don't think it's possible to make it more dynamic.

File: tests/cypress/e2e/teacher_page/customize_class_page/ind_adv_button.cy.js (hunk 1 above)
Comment (sharandaa) [reply tests/cypress/e2e/teacher_page/customize_class_page/ind_adv_button.cy.js:18]: I was able to make this dynamic! I used a for loop, I think I can maybe also do this with the date inputs on this page.

**Review by Felienne — APPROVED** [review]
Thanks for tome comments @TiBiBa and thanks for the fixes @sharandaa!! 

All works: 

image

**Comment by Felienne** [comment]
And congrats on your first merged Pull Request!! 🥳

**Pull request merged**